### PR TITLE
Assignments sort+filter

### PR DESCRIPTION
**Pull request by peter-budo**
@karllindmark 

This is your homework to check, since you are not available for chat

### CODE REVIEW

File: BF4Intel/src/main/java/com/ninetwozero/bf4intel/database/dao/assignments/AssignmentSorter.java [hunk 1]
```diff
@@ -1,43 +1,76 @@
...
-public class AssignmentSorter {
-    private static final List<String> ASSIGNMENT_TYPE = new ArrayList<String>(Arrays.asList("bronze", "silver", "gold", "sp"));
+public class AssignmentSorter extends AbstractSorter<SortedAssignmentContainer> {
+    public static final List<String> ASSIGNMENT_TYPE = new ArrayList<String>(Arrays.asList("bronze", "silver", "gold", "sp"));
+    private final Assignments assignments;
+    private final SortMode mode;
 
-    public static SortedAssignmentContainer sort(final Assignments assignments) {
-        return assignments != null ? fetchSortedAssignments(assignments) : new SortedAssignmentContainer();
+    public AssignmentSorter(final Assignments assignments, final SortMode mode) {
```
Comment (karllindmark) [inline BF4Intel/src/main/java/com/ninetwozero/bf4intel/database/dao/assignments/AssignmentSorter.java]: Just re-considering the design of the class: should we pass `SortMode mode` into the `sort()` method instead?

File: BF4Intel/src/main/java/com/ninetwozero/bf4intel/database/dao/assignments/AssignmentSorter.java [hunk 2]
```diff
@@ -1,43 +1,76 @@
...
+    public SortedAssignmentContainer sort() {
+        if (mode == SortMode.PROGRESS) {
+            return sortByProgress();
+        } else {
+            return sortByCategory();
+        }
+    }
+
+    @Override
+    protected SortedAssignmentContainer sortByProgress() {
```
Comment (karllindmark) [inline BF4Intel/src/main/java/com/ninetwozero/bf4intel/database/dao/assignments/AssignmentSorter.java:32]: Do we really want to skip assignments that we've already completed? I reckon they should just be at the bottom.

Comment (peter-budo) [inline BF4Intel/src/main/java/com/ninetwozero/bf4intel/database/dao/assignments/AssignmentSorter.java:32]: One of the things I wanted to discuss you been unavailable...

**Comment by peter-budo** [comment]
@karllindmark 
1. I finished sorting by progress
2. I added migration for 1.0.1 as previously discussed

Build success http://ci.peterscorner.co.uk/job/BF4-Intel-ON-DEMAND/5/

**Comment by karllindmark** [comment]
Side note: We need to sort out Travis sometime soon, he's being grumpy. :dancers:

**Pull request merged**